site/content/docs/cli/using-constructs.md

```diff
@@ -115,6 +115,58 @@ All resources you can create and manage using the Checkly CLI are derived from "
 You can use standard JS/TS programming to use these constructs to create the monitoring setup of your
 choice. Loops, variables, if-statements, file imports, extensions etc.
 
+### Using ECMAScript modules files
+
+If your project uses ECMAScript modules files, you can specify `type: "module"` in your `package.json` file or use `.mjs` file extensions. ECMAScript modules files can be used to create resources or as check's script dependencies.
+
+> Note that `__dirname` and `__filename` don't exist in ECMAScript so you have to adapt your setup/teardown script references using relative paths.
+> Also, as the setup/teardown script dependencies are run under a sandboxed environment, you must use CommonJS (or TypeScript project) for all the second-level dependencies for API and Browser checks.
+
+#### Example using top-level `await` supported in ECMAScript
+
+```js
+// __checks__/api.check.js
+import { ApiCheck, AssertionBuilder } from 'checkly/constructs'
+// top-level-await allowed
+const getSetupScriptName = await Promise.resolve('./utils/setup.js')
+new ApiCheck('homepage-api-check-1', {
+  name: 'Fetch Book List',
+  setupScript: {
+    // relative reference to the file (__dirname not available)
+    entrypoint: getSetupScriptName
+  },
+  request: {
+    url: 'https://danube-web.shop/api/books',
+    method: 'GET',
+    followRedirects: true,
+    skipSSL: false,
+    assertions: [
+      AssertionBuilder.statusCode().equals(200),
+      AssertionBuilder.jsonBody('$[0].id').isNotNull(),
+    ],
+  }
+})
+
+// __checks__/utils/setup.js
+import { getToken } from './auth-client'
+// top-level-await allowed
+const getEnvironmentUrl = await Promise.resolve('https://danube-web.shop')
```

Comment:
This example is contrived. Why would I return a string from a Promise? That makes no sense. We want to have examples that are based in real life.

Reply:
Yes, you are right. As in the comment above. I just wanted to show how a user could get some information from external sources (like an `axios` request). I'll change it to show something more realistic and self-explanatory.